plum/parametric.py

```diff
@@ -30,8 +30,37 @@
 
 _dispatch = Dispatcher()
 
+class ParametricTypeMeta(TypeMeta):
+    """Parametric Types must be instantiated with indexing."""
 
-class CovariantMeta(TypeMeta):
+    def __getitem__(self, *p):
+        # Type[Par1, Par2] creates a new parametric type
+
+        return self.__new__(self, *p)
+
+    def __call__(cls, *args, **kwargs):
+        # Type(arg1, arg2, kwargs) will first construct the
+        # parametric subtype T = Type[type(arg1), type(arg2)]
+        # and then call the equivalent of T(arg1, arg2, **kwargs)
+
+        if not cls.is_concrete:
+            argsT = tuple(type(arg) for arg in args)
+            T = cls[argsT]
+        else:
+            T = cls
+
+        # this is the standard type.__call__ implementation
```

Comment:
Shall we simplify this to `return type.__call__(T, *args, **kwargs)`?

Reply:
works a charm.